exp/controllers/azuremachinepool_reconciler.go

```diff
@@ -87,3 +88,17 @@ func (s *azureMachinePoolService) Delete(ctx context.Context) error {
 
 	return nil
 }
+
+func (s *azureMachinePoolService) MachinePoolMachineScopeFromAmpm(ampm *infrav1exp.AzureMachinePoolMachine) *scope.MachinePoolMachineScope {
```

Comment:
Can you help illustrate where this method is going to be used? Either by adding some more (non-breaking) changes to this PR or linking to a WIP branch?

That might help us determine whether or not there might be a better way of doing things that might not involve a method like this.

Reply:
Agreed. It could also help to add a comment to describe this function!

Reply:
Yeah, in any case we'd want to add a header comment for documenting this function as it's being exported.

@nojnhuh I paired with @bennycortese on this and explored alternatives, but this was the best solution IIRC. I believe the use case was that the AzureMachinePool controller looped through the AzureMachinePoolMachines and needed to call a function in the AMPM scope, which it didn't have access to.